### DISCUSSION TITLE
Converting PDF to JPG make image shift into blue colors

### DISCUSSION THREAD

**Discussion by saper-2** [ImageMagick/ImageMagick#3888] [Help] [answered]
Hello,

I have stuck a wall 😢 ... I need to convert pdf file into jpg - I know converting is pretty straight forward and it work good, here is command that I use (it's originally part of big script that convert pdf's to jpg and organize them):
```
convert.exe -density 96 -define pdf:use-trimbox=true "E:\_TEMP\13\TEST_0579.pdf" -resize 965034@ -colorspace RGB -quality 70 "E:\_TEMP\13\TEST_0579.jpg"
```
But I have problem that the jpg goes into "blue color", and I completely don't know how to fix that 😞 . 

Here is example, unfortunately I can not really public this pdf for tests:
![obraz](https://user-images.githubusercontent.com/8533673/123876178-31660700-d93b-11eb-8ac7-5a47f5060363.png)

Here is result of ```identify```:
```
E:\Programs\Images\ImageMagick>identify -verbose E:\_TEMP\13\TEST_0579.pdf
Image:
  Filename: E:\_TEMP\13\TEST_0579.pdf
  Format: PDF (Portable Document Format)
  Mime type: application/pdf
  Class: DirectClass
  Geometry: 1459x1176+0+0
  Resolution: 72x72
  Print size: 20.2639x16.3333
  Units: Undefined
  Colorspace: CMYK
  Type: ColorSeparation
  Endianness: Undefined
  Depth: 16/8-bit
  Channel depth:
    Cyan: 8-bit
    Magenta: 8-bit
    Yellow: 8-bit
    Black: 8-bit
  Channel statistics:
    Pixels: 1715784
    Cyan:
      min: 0  (0)
      max: 60138 (0.917647)
      mean: 1079.78 (0.0164764)
      median: 0 (0)
      standard deviation: 6450.87 (0.098434)
      kurtosis: 44.6686
      skewness: 6.61642
      entropy: 0.0646885
    Magenta:
      min: 0  (0)
      max: 54998 (0.839216)
      mean: 889.188 (0.0135681)
      median: 0 (0)
      standard deviation: 5458.18 (0.0832865)
      kurtosis: 48.5116
      skewness: 6.88051
      entropy: 0.0648305
    Yellow:
      min: 0  (0)
      max: 49087 (0.74902)
      mean: 836.332 (0.0127616)
      median: 0 (0)
      standard deviation: 4992.43 (0.0761796)
      kurtosis: 43.6071
      skewness: 6.56677
      entropy: 0.0644343
    Black:
      min: 0  (0)
      max: 65535 (1)
      mean: 793.598 (0.0121095)
      median: 0 (0)
      standard deviation: 6037.81 (0.0921311)
      kurtosis: 71.3519
      skewness: 8.36402
      entropy: 0.0525108
  Image statistics:
    Overall:
      min: 0  (0)
      max: 65535 (1)
      mean: 899.725 (0.0137289)
      median: 0 (0)
      standard deviation: 5734.82 (0.0875078)
      kurtosis: 55.1
      skewness: 7.26129
      entropy: 0.061616
  Total ink density: 347.451%
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 1459x1176+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Undefined
  Orientation: Undefined
  Profiles:
    Profile-xmp: 4984 bytes
  Properties:
    date:create: 2021-06-29T13:19:10+00:00
    date:modify: 2021-06-29T13:19:10+00:00
    dc:format: application/pdf
    pdf:HiResBoundingBox: 1459.32x1175.86+0+0
    pdf:Producer: Adobe PDF Library 15.0
    pdf:Trapped: False
    pdf:Version: PDF-1.6
    pdfx:GTS_PDFXVersion: PDF/X-4
    pdfxid:GTS_PDFXVersion: PDF/X-4
    signature: cf4928f14e7eacdded062bba91ab000a9014b159c01f3ddb14a4d690a122bf7e
    xmp:CreateDate: 2021-05-26T11:13:26+02:00
    xmp:CreatorTool: Adobe InDesign 16.1 (Macintosh)
    xmp:MetadataDate: 2021-06-29T15:10:30+02:00
    xmp:ModifyDate: 2021-06-29T15:10:30+02:00
    xmpMM:DerivedFrom: default
    xmpMM:DocumentID: xmp.id:d5973927-940c-47fc-a4ce-519f6f3e99d4
    xmpMM:InstanceID: uuid:4cb71c00-6f62-4e10-b39b-f9c72eac10a9
    xmpMM:OriginalDocumentID: xmp.did:90d72470-16b5-4497-a4d9-cbad9c555cf5
    xmpMM:RenditionClass: proof:pdf
    xmpMM:VersionID: 1
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 6.54529MiB
  Number pixels: 1.71578M
  Pixels per second: 88.2442MP
  User time: 0.016u
  Elapsed time: 0:01.019
  Version: ImageMagick 7.1.0-2 Q16 x64 2021-06-25 https://imagemagick.org
```

I have GhostScript 9.54.0 installed.

I did a googling a lot and it helped me a bit alredy (I had before yellow-ish stripes - I fixed that adding ```colorspace RGB``` option), but now the jpg run to blue 🟦 .

I'm not good with graphic - I know basic stuff: crop, cut, paste 😉 - but anything more "advanced" is over my head.

**Comment by snibgo** [comment]
What software did you use to show the PDF and JPG images?

Your PDF is encoded as CMYK. Identify shows that it has suspiciously low values in those channels.

If your PDF a wrapper for a single raster image? If so, then I suggest you extract that image with pdfimages.

You convert from CMYK to (linear) RGB. I think IM silently converts that to sRGB when writing to JPEG. It may be better to leave the image as CMYK. I suppose you have tried that, and get the yellow result. (Yellow is the complement of blue.)

IM can't see an ICC colour profile in the PDF file, so the conversion to RGB is simplistic. But the image in the PDF file may have ben encoded with an ICC profile that has not been embedded in the file. If so, then a more accurate method is to convert to a sRGB ICC profile. But that involves guessing at the CMYK profile.

**Comment by saper-2** [comment, marked as answer]
Hello,

Thanks for answer.

I view it in Adobe Reader & IrfanView/Windows Photo viewer/Paint.net.

Unfortunately I can not extract the image, I have to keep the layout unchanged in those generated "thumbnails" - those are used in warehouse for identifying printed images (and validating if they're printed OK) when they arrive from the print house.

Yes I tried already forcing or leaving to be auto-detected colorspace for in & out. I found topic in old IM forum with similar problem.

Yes, I noticed that too, there is no color profile in pdf too, since it's created in adobe I also tried few profiles that I found on net or delivered with GS - no luck.


I found via google suggestions to use directly GS, so I gave it a shoot (took me a bit of time to grasp gswin64c command line switches), and while it is a bit limited in matter of defining output size (I can tweak it by dpi resolution) but it works just almost perfect - the image looks identical, and it is capable of removing cut marks from PDF too so it's double win-win for me.

Here is for reference cmd that I'm using - for others who might encounter similar problem:
```
gswin64c.exe -dNOPAUSE -dBATCH -dUseTrimBox -sDEVICE=jpeg -r{2} -sOutputFile="{1}" "{0}"
```
This is PowerShell string to be formatted (0-input pdf, 1-output jpg file, 2-dpi resolution)

This can be !closed.
